### PR TITLE
Implement a parsedMethods option.

### LINKED ISSUE #133: koa-body should not blacklist body parsing on DELETE requests
koa-body should not blacklist DELETE method for body parsing.  

Per [RFC 7231](https://tools.ietf.org/html/rfc7231#section-4.3.5) and the HTTP 1.1 specification, it is permitted to send a body with a DELETE request.  In order to enable parsing of request bodies for DELETE methods, we have to wholesale disable strict mode, which means we're parsing it for GET and HEAD too - *this is not a good solution*.

### PR DESCRIPTION
**Pull request by damianb**
Fixes #133.

### CODE REVIEW

**Comment by damianb** [comment]
Ping @MarkHerhold for review.

File: README.md [hunk 1]
```diff
@@ -110,7 +110,8 @@ console.log('curl -i http://localhost:3000/users -d "name=test"');
 - `includeUnparsed` **{Boolean}** Toggles co-body returnRawBody option; if set to true, for form encodedand and JSON requests the raw, unparsed requesty body will be attached to `ctx.reqeust.body` using a `Symbol`, default `false`
 - `formidable` **{Object}** Options to pass to the formidable multipart parser
 - `onError` **{Function}** Custom error handle, if throw an error, you can customize the response - onError(error, context), default will throw
-- `strict` **{Boolean}** If enabled, don't parse GET, HEAD, DELETE requests, default `true`
+- `strict` **{Boolean}** ***DEPRECATED*** If enabled, don't parse GET, HEAD, DELETE requests, default `true`
+- `parsedMethods` **{String[]}** If specified, declares the methods where bodies will be parsed, default `['POST', 'PUT', 'PATCH', 'DELETE']`
```
Comment (MarkHerhold) [inline README.md]: I would change the language to read

Declares the HTTP methods where bodies will be parsed, default `['POST', 'PUT', 'PATCH', 'DELETE']`. Replaces `strict` option.

File: test/index.js [hunk 2]
```diff
@@ -460,8 +460,63 @@ describe('koa-body', () => {
           done();
         });
     });
+  });
+
+  describe('parsedMethods options',  () => {
+    beforeEach( () => {
+      //push an additional, to test the multi query
+      database.users.push({ name: 'charlike' });
```
Comment (MarkHerhold) [inline test/index.js]: Wouldn't we have to push in at least two of these to actually test if we are deleting multiple items?

File: README.md (hunk 1 above)
Comment (MarkHerhold) [reply README.md]: Disregard. We reset the "test database" before each test.

File: test/index.js (hunk 2 above)
Comment (MarkHerhold) [reply test/index.js]: Disregard. We reset the "test database" before every test.

File: README.md (hunk 1 above)
Comment (damianb) [reply README.md]: Sure.  Also, forgot that I took DELETE out of the defaults; I need to update that as well.  Give me a few ticks and I'll change & push.

Comment (damianb) [reply README.md]: Adjustment made per recommendation.

**Comment by damianb** [comment]
Okay, updated the README and also the TS definitions.  Not going to be super picky about the typescript definitions, though we *could* always throw in stricter typing of what HTTP methods actually exist.  /shrug

File: test/index.js [hunk 3]
```diff
@@ -460,8 +460,63 @@ describe('koa-body', () => {
...
+          strict: true
+        }));
+      } catch (_err) {
+        err = _err;
+      }
+
+      assert(err && err.message === 'Cannot use strict and parsedMethods options at the same time.');
+
+      done();
+    })
```
Comment (MarkHerhold) [inline test/index.js]: nitpick: missing `;`

File: index.js [hunk 4]
```diff
@@ -47,12 +47,28 @@ function requestbody(opts) {
   opts.formidable = 'formidable' in opts ? opts.formidable : {};
   opts.includeUnparsed = 'includeUnparsed' in opts ? opts.includeUnparsed : false
   opts.textLimit = 'textLimit' in opts ? opts.textLimit : '56kb';
-  opts.strict = 'strict' in opts ? opts.strict : true;
+
+  // @todo: next major version, opts.strict support should be removed
+  if (opts.strict && opts.parsedMethods) {
+    throw new Error('Cannot use strict and parsedMethods options at the same time.')
```
Comment (MarkHerhold) [inline index.js]: nitpick: I think this should technically be a `TypeError`

File: index.js [hunk 5]
```diff
@@ -47,12 +47,28 @@ function requestbody(opts) {
...
+  }
+
+  if ('strict' in opts) {
+    console.warn('DEPRECATED: opts.strict has been deprecated in favor of opts.parsedMethods.')
+    if (opts.strict) {
+      opts.parsedMethods = ['POST', 'PUT', 'PATCH']
+    } else {
+      opts.parsedMethods = ['POST', 'PUT', 'PATCH', 'GET', 'HEAD', 'DELETE']
```
Comment (MarkHerhold) [inline index.js]: My only concern with this entire PR is that we might be making a false assumption that users that set `opts.strict` to `false` only expect koa-body to parse on POST, PUT, PATCH, GET, HEAD, and DELETE methods.

This might be a problem if a user sets `opts.strict` = `false` and a custom method, e.g. `OPTIONS` or `TRACE` and expects a parsed body and then updates to this version.

Please let me know if I'm thinking about this correctly.

Comment (MarkHerhold) [reply index.js]: One thought I had was to roll this and a few other changes into a major version and worry less about these edge-cases.

File: index.js (hunk 4 above)
Comment (damianb) [reply index.js]: TypeError would not be correct as it's not a type mismatch tbh.

File: index.js (hunk 5 above)
Comment (damianb) [reply index.js]: While yes, that's definitely an edge case, we're entering edge case saloon territory; `OPTIONS` currently has no realizable use for a request body, and `TRACE` actually specifies in RFC 7231 that it "MUST NOT" possess a body.

File: test/index.js (hunk 3 above)
Comment (damianb) [reply test/index.js]: whoops

File: index.js (hunk 5 above)
Comment (MarkHerhold) [reply index.js]: Ok, makes sense. I think that this is an acceptable risk with a minor version.

**Comment by MarkHerhold** [comment]
Published as koa-body@4.1.0
